syntax = "proto3";

package spotify;

service SpotifyService {
  rpc ListarUsuarios(Empty) returns (UsuariosResponse);
  rpc ListarMusicas(Empty) returns (MusicasResponse);
  rpc ListarPlaylistsPorUsuario(UsuarioRequest) returns (PlaylistsResponse);
  rpc ListarMusicasPorPlaylist(PlaylistRequest) returns (MusicasResponse);
  rpc ListarPlaylistsPorMusica(MusicaRequest) returns (PlaylistsResponse);
}

message Empty {}

message Usuario {
  int32 id = 1;
  string nome = 2;
  int32 idade = 3;
}

message Musica {
  int32 id = 1;
  string nome = 2;
  string artista = 3;
}

message Playlist {
  int32 id = 1;
  string nome = 2;
  int32 usuario_id = 3;
}

message UsuarioRequest {
  int32 usuario_id = 1;
}

message PlaylistRequest {
  int32 playlist_id = 1;
}

message MusicaRequest {
  int32 musica_id = 1;
}

message UsuariosResponse {
  repeated Usuario usuarios = 1;
}

message MusicasResponse {
  repeated Musica musicas = 1;
}

message PlaylistsResponse {
  repeated Playlist playlists = 1;
}
